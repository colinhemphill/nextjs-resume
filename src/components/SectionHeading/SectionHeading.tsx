import { faCircle, IconDefinition } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import React from 'react';
import { Heading, HeadingProps } from '../Heading/Heading';

interface SectionHeadingProps {
  icon?: IconDefinition;
  level?: HeadingProps['level'];
  text: string;
}

export const SectionHeading: React.FC<SectionHeadingProps> = (props) => {
  const { icon, level = 3, text } = props;

  return (
    <Heading level={level}>
      <div className="flex items-center gap-2">
        {icon && (
          <span className="fa-layers fa-fw hidden md:flex">
            <FontAwesomeIcon className="text-neutral-12" icon={faCircle} />
            <FontAwesomeIcon
              aria-hidden
              className="text-neutral-1"
              icon={icon}
              transform="shrink-8"
            />
          </span>
        )}
        <div>{text}</div>
      </div>
    </Heading>
  );
};
