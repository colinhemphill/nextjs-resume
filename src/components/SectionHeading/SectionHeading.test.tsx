import { render, screen } from '@test-utils';
import { describe, expect, test } from 'vitest';
import { SectionHeading } from './SectionHeading';

describe('<SectionHeading />', () => {
  test('Snapshot', async () => {
    const { container } = render(<SectionHeading text="Section Heading" />);
    expect(container).toMatchSnapshot();
  });

  test('Renders a heading', async () => {
    render(<SectionHeading text="Section Heading" />);

    const heading = screen.getByRole('heading', {
      level: 3,
      name: /Section Heading/i,
    });

    expect(heading).toBeDefined();
  });
});
