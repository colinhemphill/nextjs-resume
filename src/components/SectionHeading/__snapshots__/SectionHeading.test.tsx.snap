// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<SectionHeading /> > Snapshot 1`] = `
<div>
  <h3
    class="font-bold text-neutral-12 text-xl md:text-2xl"
  >
    <div
      class="flex items-center gap-2"
    >
      <div>
        Section Heading
      </div>
    </div>
  </h3>
</div>
`;
